flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java

```diff
@@ -212,4 +217,59 @@ public static List<String> loadAndModifyConfiguration(String[] args, String cmdL
         return ConfigurationUtils.convertConfigToWritableLines(
                 configuration, modifiableClusterConfiguration.flattenConfig());
     }
+
+    public static List<String> migrateLegacyConfigurationToStandardYaml(
+            String[] args, String cmdLineSyntax) throws FlinkParseException {
+        final CommandLineParser<ClusterConfiguration> commandLineParser =
+                new CommandLineParser<>(new ClusterConfigurationParserFactory());
+
+        final ClusterConfiguration clusterConfiguration;
+
+        try {
+            clusterConfiguration = commandLineParser.parse(args);
+        } catch (FlinkParseException e) {
+            LOG.error("Could not parse the command line options.", e);
+            commandLineParser.printHelp(cmdLineSyntax);
+            throw e;
+        }
+
+        checkState(
+                new File(
+                                clusterConfiguration.getConfigDir(),
+                                GlobalConfiguration.LEGACY_FLINK_CONF_FILENAME)
+                        .exists());
+        Configuration configuration =
+                GlobalConfiguration.loadConfiguration(clusterConfiguration.getConfigDir(), null);
+
+        Configuration standardYamlConfig = new Configuration(true);
+        standardYamlConfig.addAll(configuration);
+
+        // Below config options type is String or List<String>, but it will be used as Collection
+        // type, so we should explicitly reset the value
+        configuration
```

Comment:
Is this really needed?
It's weird that we have to handle some of the config options in a special way.